hnn_core/cell.py

```diff
@@ -322,7 +339,13 @@ def __init__(self, name, pos, sections, synapses, topology, sect_loc,
         if gid is not None:
             self.gid = gid  # use setter method to check input argument gid
 
-        self._update_end_pts()
+        # Store the tree representation of the cell
+        self.cell_tree = cell_tree
+
+        # self._update_end_pts()  # Old implementation
+        self.update_end_pts()  # New implementation
```

Comment:
do you still need the method `_update_end_pts` ? We should consistently update everywhere and remove the old method if not needed